src/Npgsql/ConnectorPool.cs

```diff
@@ -111,6 +115,7 @@ internal ConnectorPool(NpgsqlConnectionStringBuilder settings, string connString
                 : settings.ToStringWithoutPassword();
 
             _pruningInterval = TimeSpan.FromSeconds(Settings.ConnectionPruningInterval);
+            _pruningTimerCallback = PruneIdleConnectors;
```

Comment:
This could be made even better. Just change `PruneIdleConnectors` to be static and pass the current instance to the timer on line 268. So there will be only one instance of the delegate pointing to the method.

Reply:
Done, it doesn't win any beauty contests but on perf it's optimal :)